### ISSUE TITLE
Fatal Error: Failed to execute _main_ when clicking on "Run"

### ISSUE THREAD

**Issue by XenSide** [garashchenko/mystic-why#6]
![Discord_zIZcktkQyg](https://user-images.githubusercontent.com/20782583/132089662-a4abc0cf-176e-4239-a272-c82ad1749c51.png)

As soon as I click Run (with any selection both for effects, zones or colors) I get a Fatal Error: Failed to execute _main_

Unsure if there's something I was supposed to do that I skipped?

**Comment by garashchenko** [comment]
Thanks for your feedback! I will look into this next week.
Do you use the latest .exe file from Releases?

**Comment by XenSide** [comment]
I did!

Big disclaimer, I'm no python expert, but I think I found a workaround:

I worked around the issue by downloading the source and it appears the problem was with how the python package was built, I worked around this by adding:

```
import os, sys, inspect

cmd_folder = os.path.realpath(os.path.abspath(os.path.split(inspect.getfile( inspect.currentframe() ))[0]))
if cmd_folder not in sys.path:
    sys.path.insert(0, cmd_folder)


subfolders = ["gui","hooks","effects","core","common"]
for subfolder in subfolders:
    cmd_subfolder = os.path.realpath(os.path.abspath(os.path.join(os.path.split(inspect.getfile( inspect.currentframe() ))[0],subfolder)))
    if cmd_subfolder not in sys.path:
        sys.path.insert(0, cmd_subfolder)
```

To the __main__.py file before the import and then refactoring every import in any file by removing the root folder name "mystic_why." (so basically a replace "mystic_why." with "") AND adding `__init__.py` files in every single subfolder.
From my understanding this way both the main folder and all the subfolders get added to the PythonPath.

This made the whole project work directly by starting it with Python

**BUT**

Now when I go press Run the program closes and a "**DeviceNotFound**" exception gets thrown so I've looked around a bit and in OpenRGB my motherboard is listed as MS-7B85 which is not present in your readme, could it be that it's a model that you didn't know existed and thus is not whitelisted or something among those lines?

**Comment by XenSide** [comment]
Correct me if I'm wrong, you check for a "MYSTIC LIGHT " string when checking HIDs, right? Because apparently my motherboard does not have a mystic light string in its HID.

The HID call you use extracts this: `{'path': b'\\\\?\\hid#vid_1462&pid_7b85#7&391bdf99&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}', 'vendor_id': 5218, 'product_id': 31621, 'serial_number': 'A02018061512', 'release_number': 1, 'manufacturer_string': 'MSI', 'product_string': 'PRO CARBON 12', 'usage_page': 1, 'usage': 0, 'interface_number': -1}` 
In my case


EDIT:

I'm pretty sure my dumb-ass self just bricked its own mobo RGB controller

EDIT2: 
Clearing CMOS and leaving the system unpowered for a bit solved the issue, so no, I did not brick the controller but at the same time something doesn't work with my MOBO and mystic-why.
Reading around people with bricked MOBOs it seems like most mystic light mobos use 185 byte protocols while mine uses 162 (Gaming Pro Carbon ac).

**So basically, there probably was a reason why my product string didn't have "MYSTIC LIGHT" in it lol**

Summing this up, is the problem the different byte size protocol? and if so, is there any way you can make it compatible?

**Comment by garashchenko** [comment]
Thanks for diving that deep in the problem! :) I don’t have a PC at the moment so I can’t check for sure, but yeah, there are different message protocols and my project implements only 185 byte one (and its direct mode). 

Your motherboard seems to use an older version of message type and I guess it should work with MSIRGB project - https://github.com/ixjf/MSIRGB

**Comment by XenSide** [comment]
Unfortunately no, MSIRGB does not work for the B450 Gaming Pro Carbon AC

OpenRGB does but is unable to extend the Jrainbow headers to more than one LED (or atleast, in the GUI the option is grayed out) meaning any effect applies to the whole fan at once like it would in a normal non-**A**RGB fan
